backend/src/routes/api/gpu/gpuUtils.ts

```diff
@@ -1,9 +1,16 @@
-import { KubeFastifyInstance, PrometheusResponse } from '../../../types';
+import {
+  MachineAutoscalerList,
+  GPUInfo,
+  KubeFastifyInstance,
+  PrometheusResponse,
+  MachineSet,
+  gpuScale,
+} from '../../../types';
 import { V1PodList } from '@kubernetes/client-node';
 import https from 'https';
 import * as fs from 'fs';
 
-export const getGPUNumber = async (fastify: KubeFastifyInstance): Promise<[boolean, number]> => {
+export const getGPUNumber = async (fastify: KubeFastifyInstance): Promise<GPUInfo> => {
   let maxGpuNumber = 0;
   let areGpusConfigured = false;
   const gpuPodList = await fastify.kube.coreV1Api
```

Comment:
If the only gpus are on a machine pool that is scaled to 0, you're not going to have any pods, the daemonset `nvidia-dcgm-exporter` would have scaled them to 0.  If you're looking for the daemonset you should query for that.  You could alternatively query for the subscription `nvidia-gpu-operator` and check its `csv` the same way the tiles do.  You probably need to look into the recommended way to do this.

Reply:
FWIW, he does a check if he found scaling information he doesn't need the pod information. [code link](https://github.com/opendatahub-io/odh-dashboard/pull/573/files#diff-fe048868b84dce849bd698d4f237f445ecc70661f9ca1842d9a9429b4c2c37feR50)

Reply:
sorry, was on looking at an older version before your last commit